python/lsst/pipe/tasks/characterizeImage.py

```diff
@@ -336,13 +342,13 @@ def __init__(self, butler=None, refObjLoader=None, schema=None, **kwargs):
 
     def runQuantum(self, butlerQC, inputRefs, outputRefs):
         inputs = butlerQC.get(inputRefs)
-        if 'exposureIdInfo' not in inputs.keys():
-            inputs['exposureIdInfo'] = ExposureIdInfo.fromDataId(butlerQC.quantum.dataId, "visit_detector")
+        if 'idGenerator' not in inputs.keys():
+            inputs['idGenerator'] = self.config.idGenerator.apply(butlerQC.quantum.dataId)
```

Comment:
When would `idGenerator` be in `inputs.keys()`? I don't see how that could happen.

Reply:
Me neither, just replicating the existing logic.

Reply:
Digging into the history, this is a holdover from the gen2 `adaptArgsAndRun`, and only `processCcdWithFakes` has similar logic (added at a similar time, likely one cargo culted from the other). That `if` looks entirely unnecessary now. Don't need to change it here, since I plan to make `characterizeImage` go away with the new `CalibrateImage`.